src/deck/drivers/src/aideck.c

```diff
@@ -53,13 +52,24 @@ static uint8_t byte;
 //Uncomment when NINA printout read is desired from console
 //#define DEBUG_NINA_PRINT
 
-#ifdef DEBUG_NINA_PRINT
 static void NinaTask(void *param)
 {
     systemWaitStart();
+  vTaskDelay(M2T(3000));
+  if (espDeckFlasherCheckVersionAndBoot() == false)
+  {
+    DEBUG_PRINT("ESP32 not booted.");
+    while (1)
+    {
+      vTaskDelay(portMAX_DELAY);
+    }
```

Comment:
So we loop forever here?

Reply:
This is similar to the lighthouse deck flashing protocol, to stay in the bootloader if the firmware hash is wrong (`lighthouseCoreTask`). In the ESP case we need no such safety as we are not touching the ESP ROM bootloader. On the other hand, is there a point in continuing with the UART readout if your checksum failed and you are running faulty firmware on the ESP?